src/components/Accounts/Actions/ActionModal.js

```diff
@@ -336,9 +345,16 @@ class ActionModal extends React.Component {
     } = getState();
 
     const selected = calculateSelectedAmount(accounts, this.isRefundAction(action), feeFineActions);
-    const ownerOptions = owners.filter(o => o.owner !== 'Shared').map(o => ({ value: o.id, label: o.owner }));
+    const ownersServicePoint = [];
+    owners.forEach((item) => {
+      if (item.servicePointOwner.filter(s => s.value === curServicePoint.id).length > 0) {
+        ownersServicePoint.push(item);
+      }
+    });
+    const ownerOptions = ownersServicePoint.filter(o => o.owner !== 'Shared').map(o => ({ value: o.id, label: o.owner }));
```

Comment:
There are a lot of these kinds of list filters throughout the fee-fine codebase. It would be really helpful if we had comments explaining what they do, and what they're for, e.g. 
```
// only fee-fine owners affiliated with the current SP should be displayed.
// filter out all the others. 
const ownersServicePoint = [];
...
```
as it is, we have to read the code to figure out what's happening, and then find the associated ticket to figure out why.